binary.js

```diff
@@ -5,17 +5,46 @@ var testArrayGanjil = [3, 31, 89, 53, 53, 85, 77, 21, 55]
 
 function ownSort(arr) {
   // Your sorting code
-  return arr
+
+  for (let i = 1; i < arr.length; i++) {
+    let temp = arr[i];
+
+    for (let j = i; j > 0; j--) {
+      if (arr[j-1] > arr[j]) {
+        arr[j] = arr[j-1];
+        arr[j-1] = temp;
+      }
+    }
+  }
+  return arr;
 }
 
-function binarySearch (search, array) {
+
+function binary_search (search, array) {
   // Your searching code
-  return 0;
-}
+  let width = array.length
+
+  let center = (width-1)/2;
+  while (width > 0) {
+    if (search > array[center]) {
+      center+=Math.floor((center/2)+1)
```

Comment:
Indentasinya dirapihkan yaa
```suggestion
      center += Math.floor((center/2)+1)
```